### PR TITLE
Make the whiteout file check more precise

### PR DESCRIPTION
**Pull request by angelhof**
Addresses #42

I tried to create a test for this that differentiates between a real character device file being created and the whiteout file but I found it really hard (creating an actual device file requires sudo and using fakeroot does not create an actual character device file). 

Any ideas on how to create a test would be pretty useful :)

### CODE REVIEW

File: try [hunk 1]
```diff
@@ -205,6 +205,12 @@ $changed_files
 EOF
 }
 
+## Checks if a file is an overlayfs whiteout file
+is_whiteout_file()
+{
+    local file="$1"
```
Comment (mgree) [inline try:211]: `local` is a `bash`ism/unspec in POSIX. Better to just use `$1` throughout.

**Comment by mgree** [comment]
Ah... annoying. One possibility would be to define a small fuse filesystem that fakes such a device. Alternatively, we could have certain tests marked as requiring `sudo`, turn those off by default, but run them in CI.

File: try (hunk 1 above)
Comment (angelhof) [reply try:211]: I removed the local, but kept the renaming to `file` for better readability. If you meant to completely remove that too, let me know and I can do it.

**Comment by angelhof** [comment]
Hmm, I investigated this and it still does not look easy:
- `mknod` in try gives `Permission denied` even though it works outside (`sudo mknod my-null c 0 0`). We will need to investigate why this happens.
- fuse can be used to create character files, but I haven't figured out how to create a character file in try using it, will keep investigating

**Comment by mgree** [comment]
Well, we _do_ have some tests on `rm`... maybe we can add a test to make sure a "deleted" shows up in a summary? That way we know that we're detecting whiteouts correctly (without confirming that non-whiteouts are left alone). Then we can merge and leave an issue to test with a real, non-`0,0` character device later?

**Pull request merged**